java/runtime/src/main/java/org/ray/runtime/AbstractRayRuntime.java

```diff
@@ -71,14 +71,14 @@ public AbstractRayRuntime(RayConfig rayConfig) {
   @Override
   public <T> RayObject<T> put(T obj) {
     UniqueId objectId = UniqueIdUtil.computePutId(
-        workerContext.getCurrentTask().taskId, workerContext.nextPutIndex());
+        workerContext.getCurrentTaskId(), workerContext.nextPutIndex());
```

Comment:
Hmm, I just realized that this may interact poorly with fault tolerance. If an object created on a separate thread is lost, then the backend will probably crash because the object won't have any lineage. We don't need to address it in this PR, but just something to keep in mind...

Reply:
Good point. Maybe we should change [this check](https://github.com/ray-project/ray/blob/ad2287ebe9bb9b02f91909666b8cd86569203bf3/src/ray/raylet/node_manager.cc#L1684) to a warning?

Reply:
Yeah, eventually we probably should, but it's still useful right now for debugging purposes.